test/dispatchers/dispatcherTests.ts

```diff
@@ -2,72 +2,100 @@
 
+      eventCallbackCalls = 0;
+      dispatcher.setCallback(callbackSet, callback);
       TestMethods.triggerFakeUIEvent("click", d3document);
-      assert.isTrue(callbackWasCalled, "connected correctly (callback was called)");
+      assert.strictEqual(eventCallbackCalls, 1, "eventCallback is called after connect");
+      assert.strictEqual(callbackCalls, 1, "callback is called after setting callback");
 
-      (<any> dispatcher)._disconnect();
-      callbackWasCalled = false;
+      callbackCalls = 0;
+      dispatcher.setCallback(callbackSet, callback);
       TestMethods.triggerFakeUIEvent("click", d3document);
-      assert.isTrue(callbackWasCalled, "didn't disconnect while dispatcher had listener");
+      assert.strictEqual(callbackCalls, 1, "callback is only called once even if same callback is added twice");
 
-      callbackSet.delete(callback);
-      (<any> dispatcher)._disconnect();
-      callbackWasCalled = false;
+      callbackCalls = 0;
+      dispatcher.unsetCallback(callbackSet, callback);
       TestMethods.triggerFakeUIEvent("click", d3document);
-      assert.isFalse(callbackWasCalled, "disconnected when dispatcher had no listeners");
+      assert.strictEqual(callbackCalls, 0, "callback is not called after unsetting");
     });
 
-    it("_setCallback()", () => {
-      let dispatcher = new Plottable.Dispatcher();
-      let callbackSet = new Plottable.Utils.CallbackSet<Function>();
-
-      let callbackWasCalled = false;
-      let callback = () => callbackWasCalled = true;
-
-      (<any> dispatcher)._setCallback(callbackSet, callback);
-      callbackSet.callCallbacks();
-      assert.isTrue(callbackWasCalled, "callback was called after setting with _setCallback()");
+    it("unsets one callback doesn't disconnect other callbacks", () => {
+      let callback2 = () => callbackCalls++;
+      dispatcher.listenToEvent("click", eventCallback);
+      dispatcher.setCallback(callbackSet, callback);
+      dispatcher.setCallback(callbackSet, callback2);
+      TestMethods.triggerFakeUIEvent("click", d3document);
+      assert.strictEqual(eventCallbackCalls, 1, "eventCallback is called after connect");
+      assert.strictEqual(callbackCalls, 2, "callback is called twice for 2 callbacks");
 
-      (<any> dispatcher)._unsetCallback(callbackSet, callback);
-      callbackWasCalled = false;
-      callbackSet.callCallbacks();
-      assert.isFalse(callbackWasCalled, "callback was removed by calling _unsetCallback()");
+      callbackCalls = 0;
+      eventCallbackCalls = 0;
+      dispatcher.unsetCallback(callbackSet, callback);
+      TestMethods.triggerFakeUIEvent("click", d3document);
+            assert.strictEqual(eventCallbackCalls, 1, "eventCallback is still called");
+      assert.strictEqual(callbackCalls, 1, "callback is called exactly once");
+      dispatcher.unsetCallback(callbackSet, callback2);
     });
   });
 });
+
+module Mocks {
+  export class BasicDispatcher extends Plottable.Dispatcher {
```

Comment:
The main thing I worry about is that this might be disconnected from the actual child class implementations.  Do you think we should work the the mock here or do you think we should use a different method?